src/analyze/replacements.ts

```diff
@@ -60,59 +104,92 @@ export async function runReplacements(
     return result;
   }
 
+  // Load custom manifests
+  const customReplacements = await loadCustomManifests(options?.manifest);
+
+  // Combine custom and built-in replacements
+  const allReplacements = [
+    ...customReplacements,
+    ...replacements.all.moduleReplacements
+  ];
+
   for (const name of Object.keys(packageJson.dependencies)) {
-    const replacement = replacements.all.moduleReplacements.find(
+    // Find replacement (custom replacements take precedence due to order)
+    const replacement = allReplacements.find(
       (replacement) => replacement.moduleName === name
     );
 
     if (!replacement) {
       continue;
     }
 
-    if (replacement.type === 'none') {
-      result.messages.push({
-        severity: 'warning',
-        score: 0,
-        message: `Module "${name}" can be removed, and native functionality used instead`
-      });
-    } else if (replacement.type === 'simple') {
-      result.messages.push({
-        severity: 'warning',
-        score: 0,
-        message: `Module "${name}" can be replaced. ${replacement.replacement}.`
-      });
-    } else if (replacement.type === 'native') {
-      const enginesNode = packageJson.engines?.node;
-      let supported = true;
-
-      if (replacement.nodeVersion && enginesNode) {
-        supported = isNodeEngineCompatible(
-          replacement.nodeVersion,
-          enginesNode
-        );
+    // Handle each replacement type using the same logic for both custom and built-in
+    switch (replacement.type) {
```

Comment:
basically everything inside this switch should really be what it was before

whether we use an if or a switch, i don't mind. but the messages shouldn't need to be refactored

if they do, let me know why